test/lightning/workflow_versions_test.exs

```diff
@@ -46,10 +46,115 @@ defmodule Lightning.WorkflowVersionsTest do
       assert {:error, :invalid_input} =
                WorkflowVersions.record_version(wf, @a, "web")
     end
+
+    test "does not insert duplicate when hash is same as latest" do
```

Comment:
Hmm, not totally sure on these behaviours.

I think the correct behaviour is:

* If the source is the same, and the hash is the same, then it's basically a no-op. No new row inserted, and I guess we don't update the timestamp.
* If the source is the different but the hash is the same, then we must push a new row. So I think the assertion around line 63 is wrong
* If the source is the same and the hash is different, we squash (so still no new row)

I suppose the version history we're comparing doesn't include the source - so maybe I'm just getting misled by the test here?

Reply:
Thinking about this I really think you need to include the source in the version history - so `['cli:aaaaa', 'app:bbbb']`

Reply:
Aaah, that's good to know. Let me make the change